javalib/src/main/scala/java/util/PriorityQueue.scala

```diff
@@ -137,16 +210,23 @@ class PriorityQueue[E] private (
 
   def iterator(): Iterator[E] = {
     new Iterator[E] {
-      private[this] var inner: js.Array[E] = PriorityQueue.this.inner
+      private[this] var innerJS: js.Array[E] = PriorityQueue.this.innerJS
+      private[this] var innerWasm: Array[AnyRef] = PriorityQueue.this.innerWasm
```

Comment:
Ouch, this is going to be problematic. It forces reading of both fields from the enclosing class. That means the optimizer and dead code elimination will not be able to remove the field that is unused on the given platform.

Reply:
Instead of having two fields, can we just do something like:
```scala
var inner: AnyRef = if (isWasm) ... else ...
@inline def innerWasm = inner.asInstanceOf[...]
@inline def innerJS = inner.asInstanceOf[...]
```

Reply:
That's one possibility. But it does incur a cast down to `Array[AnyRef]` every time we use it on Wasm. (the cast to `js.Array` is free).